src/main/kotlin/org/jetbrains/bio/big/Bed.kt

```diff
@@ -66,67 +78,109 @@ data class BedEntry(
             omitEmptyStrings: Boolean = false
     ): ExtendedBedEntry {
 
-        check(fieldsNumber in 3..12) { "Fields number expected 3..12, but was $fieldsNumber" }
+        check(fieldsNumber in 3..12) { "Fields number expected in range 3..12, but was $fieldsNumber" }
 
-        // This impl parsed only BED9 and adds 9..12 fields to 'rest' string
         val limit = fieldsNumber.toInt() - 3 + 1
-        val it = when {
-            rest.isEmpty() -> emptyArray<String>().iterator()
-            omitEmptyStrings -> Splitter.on(delimiter).trimResults().omitEmptyStrings().limit(limit).split(rest).iterator()
-            else -> rest.split(delimiter, limit = limit).iterator()
+        val fields = when {
+            rest.isEmpty() -> emptyList<String>()
+            omitEmptyStrings -> Splitter.on(delimiter).trimResults().omitEmptyStrings().limit(limit).split(rest)
+            else -> rest.split(delimiter, limit = limit)
+        }.toList()
+        if (fields.size < fieldsNumber - 3) {
+            throw BedEntryUnpackException(this, (fields.size + 3).toByte(), "field is missing")
         }
+        val it = fields.iterator()
 
-        // If line is shorter than suggested fields number do not throw an error
-        // it could be ok for default behaviour, when user not sure how much fields
-        // do we actually have
-        val name = if (fieldsNumber >= 4 && it.hasNext()) it.next() else "."
+        val name = if (fieldsNumber >= 4) it.next() else "."
         val score = when {
-            fieldsNumber >= 5 && it.hasNext() -> {
+            fieldsNumber >= 5 -> {
                 val chunk = it.next()
-                if (chunk == ".") 0 else chunk.toInt()
+                if (chunk == ".") {
+                    0
+                } else {
+                    chunk.toIntOrNull()
+                            ?: throw BedEntryUnpackException(this, 4, "score value $chunk is not an integer")
+                }
             }
             else -> 0
         }
-        val strand = if (fieldsNumber >= 6 && it.hasNext()) it.next().first() else '.'
+        val strand = if (fieldsNumber >= 6) it.next().firstOrNull() ?: '.' else '.'
```

Comment:
` '.' else '.'` could we avoid this code duplication?

Reply:
Bothers me too, but it's tricky. These dots are somewhat different. One `'.'` is a result of a missing field (`fieldsNumber < 6`), the other requires that `fieldsNumber >= 6`, advances the iterator and receives an empty string from it. I can't think of a nice way to do it.

Reply:
ok